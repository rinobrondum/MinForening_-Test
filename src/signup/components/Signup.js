import React, {useEffect, useState} from 'react'
import {connect} from 'react-redux'
import {compose} from 'recompose'
import {createStructuredSelector} from 'reselect'
import {Switch, Route, Redirect} from 'react-router-dom'
import {withRouterParams} from 'lib/hoc'
import {getIsAuthenticated} from 'signup/user'
import {fetchDummies} from 'signup/actions'
import {Page, Box} from 'components'
import {LogoWithTagline} from 'components/icons'
import UserList from './UserList'
import User from './User'
import Invalid from './Invalid'
import Done from './Done'
import { globalState } from 'app/components/App'

export const Signup = ({
  fetchDummies,
  match: {path},
  params: {email, password},
  location: {pathname},
}) => {
  const [invalid, setInvalid] = useState(!email && !password)
  const [ready, setReady] = useState(false)
  const [whiteLabelData, setWhiteLabelData] = useState(null)
  
  useEffect(() => {

    new Promise(async (resolve, reject) => {
      if (!whiteLabelData) {
        await fetch('/paths/paths.json?q=' + Math.floor(Math.random() * 10000000).toString())
          .then((res) => res.json())
          .then((data) => {
            setWhiteLabelData(data);
            globalState.setState('apiMyOrgUrl', data.appSettings.apiMyOrgUrl);
            globalState.setState('domainSettings', data.domainSettings);
        });
      }
     
      if (whiteLabelData) {
        if (email && password) {
          return fetchDummies({email, password, resolve, reject})
        } else {
          setReady(true)
          setInvalid(true)
        }
      }
    })
      .then(() => setReady(true))
  }, [email, password, fetchDummies, whiteLabelData])
  
  return ready ? (
    invalid && pathname !== '/register/invalid' ? (
      <Redirect to="/register/invalid" />
    ) : (
      <>
        {whiteLabelData &&
          <Page>
            <Box my={4}>
              {
                whiteLabelData && whiteLabelData.logoTexts && 
                  <LogoWithTagline height="50px" width="100%" whiteLabelData={whiteLabelData}/>
              }
            </Box>

            <Switch>
              <Route exact path={path} component={UserList} />
              <Route exact path={`${path}/done`} render={(props) => <Done {...props} whiteLabelData={whiteLabelData}/>} />
              <Route exact path={`${path}/invalid`} render={(props) => <Invalid {...props} whiteLabelData={whiteLabelData}/>} />
              <Route path={`${path}/:id`} component={User} />
            </Switch>
          </Page>
        }
      </>
    )
  ) : null
}

const enhancer = compose(
  withRouterParams,
  connect(
    createStructuredSelector({
      isAuthenticated: getIsAuthenticated,
    }),
    {
      fetchDummies: fetchDummies.requested,
    }
  )
)

export default enhancer(Signup)
