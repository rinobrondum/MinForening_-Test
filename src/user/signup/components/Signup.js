import {Component} from 'react'
import {connect} from 'react-redux'
import {compose} from 'recompose'
import {createStructuredSelector} from 'reselect'
import {withRouterParams} from 'lib/hoc'
import {getClubInformation} from '../actions'
import {getIsFetching} from '../selectors'
import {logout} from 'authentication'

export class Signup extends Component {
  componentDidMount() {
    const {
      logout,
      getClubInformation,
      params: {clubToken},
    } = this.props

    logout()
  
    if (clubToken) {

      getClubInformation({clubToken})
    }
  }

  render() {
    return null
  }
}

const enhancer = compose(
  withRouterParams,
  connect(
    createStructuredSelector({
      isFetching: getIsFetching,
    }),
    {
      logout,
      getClubInformation: getClubInformation.requested,
    }
  )
)

export default enhancer(Signup)
