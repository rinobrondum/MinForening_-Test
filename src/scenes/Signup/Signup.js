import React, {useEffect} from 'react'
import {connect} from 'react-redux'
import {Flow as SignupFlow} from 'user/signup/components'
import {Flex} from 'components'
import {logout} from 'authentication'

const Signup = ({logout, whiteLabelData}) => {
  useEffect(() => {
    logout()
    // eslint-disable-next-line
  }, [])

  return (
    <Flex
      alignItems="center"
      flexDirection="column"
      justifyContent="center"
      flex="1"
      p={3}
    >
      <SignupFlow whiteLabelData={whiteLabelData}/>
    </Flex>
  )
}

const enhancer = connect(null, {logout})

export default enhancer(Signup)
