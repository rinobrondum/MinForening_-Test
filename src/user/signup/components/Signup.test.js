import React from 'react'
import { shallow } from 'enzyme'
import { Signup } from './Signup'

const setup = overrideProps => {
  const props = {
    ...overrideProps,
    logout: jest.fn(),
    getClubInformation: jest.fn(),
    params: {
      clubToken: '123abc',
    },
  }

  const wrapper = shallow(<Signup {...props} />)

  return {
    wrapper,
    props,
  }
}

describe('<Signup />', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  it('calls logout after mount', () => {
    const {
      wrapper,
      props: { logout },
    } = setup()

    expect(logout).toHaveBeenCalled()
  })

  it('call getClubInformation if clubToken is present', () => {
    const {
      wrapper,
      props: {
        getClubInformation,
        params: { clubToken },
      },
    } = setup()

    expect(getClubInformation).toHaveBeenCalledWith({ clubToken })
  })
})
