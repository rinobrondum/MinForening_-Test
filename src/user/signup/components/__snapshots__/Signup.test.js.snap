// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> renders without crashing 1`] = `ShallowWrapper {}`;
